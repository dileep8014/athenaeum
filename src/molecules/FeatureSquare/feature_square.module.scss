.feature-square {
  color: colors('neutral-8');
  padding: ru(2);
  text-align: center;
}
