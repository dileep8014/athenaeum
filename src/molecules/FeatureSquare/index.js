import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './feature_square.module.scss';
import colors from 'atoms/Color/colors.scss';

import Icon from 'atoms/Icon';
import Text from 'atoms/Text';

function FeatureSquare(props) {
  const {
    children,
    className,
    backgroundColor,
    header,
    subheader,
    icon
  } = props;

  const classes = [
    styles['feature-square'],
    className,
    backgroundColor && colors['neutral-8'],
    colors[`background-${backgroundColor}`],
  ];

  return (
    <div className={classnames(...classes)}>
      <Icon icon={icon} />
      <Text type={4}>{ header }</Text>
      <Text type={6}>{ subheader }</Text>
    </div>
  );
}


FeatureSquare.propTypes = {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,
};

FeatureSquare.defaultProps = {
  // Place any default props here.
};

export default FeatureSquare;
