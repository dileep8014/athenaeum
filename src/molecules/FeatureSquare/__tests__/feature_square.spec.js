import React from 'react';
import renderer from 'react-test-renderer';

import FeatureSquare from 'molecules/FeatureSquare';

describe('<FeatureSquare />', () => {
  it('matches snapshot', () => {
    const props = {
      header: 'Header!',
      subheader: 'Subheader!!!',
      buttonText: 'CLICK SOMETHING ALREADY!',
      icon: 'xIcon'
    };

    const actual = renderer.create(<FeatureSquare {...props} />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
