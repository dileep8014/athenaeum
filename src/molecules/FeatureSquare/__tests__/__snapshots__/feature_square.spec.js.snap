// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeatureSquare /> matches snapshot 1`] = `
<div
  className="feature-square background-undefined"
>
  <span
    className="icon-wrapper icon"
    onClick={undefined}
    style={
      Object {
        "height": undefined,
        "width": "100px",
      }
    }
  >
    <img
      alt="xIcon"
      className="img"
      src="https://storage.googleapis.com/pg-static-assets/svgs/xIcon.svg"
    />
  </span>
  <p
    className="primary-3 type-a-5-bold inherit"
  >
    Header!
  </p>
  <p
    className="primary-3 subheader type-b-8-medium inherit"
  >
    Subheader!!!
  </p>
  <div
    className="layout button-wrapper"
    id={undefined}
    style={undefined}
  >
    <div
      className="col col-sm-6 border-neutral-6"
      id={undefined}
      onClick={undefined}
      style={undefined}
    >
      <button
        className="button outline button"
        disabled={undefined}
        onClick={undefined}
        type="button"
      >
        CLICK SOMETHING ALREADY!
      </button>
    </div>
  </div>
</div>
`;
